pkg/controller/certificaterequests/controller.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificaterequests
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/utils/clock"
+
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	cmlisters "github.com/jetstack/cert-manager/pkg/client/listers/certmanager/v1alpha1"
+	controllerpkg "github.com/jetstack/cert-manager/pkg/controller"
+	"github.com/jetstack/cert-manager/pkg/issuer"
+	logf "github.com/jetstack/cert-manager/pkg/logs"
+	"github.com/jetstack/cert-manager/pkg/metrics"
+)
+
+const (
+	ControllerName = "certificaterequests"
+)
+
+var keyFunc = controllerpkg.KeyFunc
+
+type controller struct {
+	helper        issuer.Helper
+	issuerFactory issuer.IssuerFactory
+
+	// clientset used to update cert-manager API resources
+	cmClient cmclient.Interface
+	kClient  kubernetes.Interface
+
+	// To allow injection for testing.
+	syncHandler func(ctx context.Context, key string) error
+
+	issuerLister             cmlisters.IssuerLister
+	clusterIssuerLister      cmlisters.ClusterIssuerLister
+	certificateRequestLister cmlisters.CertificateRequestLister
+
+	queue   workqueue.RateLimitingInterface
+	metrics *metrics.Metrics
+
+	// logger to be used by this controller
+	log logr.Logger
+
+	// used for testing
+	clock clock.Clock
+	// used to record Events about resources to the API
+	recorder record.EventRecorder
+
+	// if addOwnerReferences is enabled then the controller will add owner references
+	// to the secret resources it creates
+	addOwnerReferences bool
+}
+
+// Register registers and constructs the controller using the provided context.
+// It returns the workqueue to be used to enqueue items, a list of
+// InformerSynced functions that must be synced, or an error.
+func (c *controller) Register(ctx *controllerpkg.Context) (workqueue.RateLimitingInterface, []cache.InformerSynced, error) {
+	// construct a new named logger to be reused throughout the controller
+	c.log = logf.FromContext(ctx.RootContext, ControllerName)
+
+	// create a queue used to queue up items to be processed
+	c.queue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Second*5, time.Minute*30), ControllerName)
```

Comment:
We can use the default rate limiter here I think 😄 (`controllerpkg.DefaultRateLimiter` or something)

Reply:
These higher back off times are typically only used for the ACME issuer